UPGRADING.md

```diff
@@ -6,6 +6,8 @@ upgrading IRIDA that cannot be automated.
 
 19.01 to 19.04
 --------------
+* This upgrade makes schema changes to the databases and cannot be parallel deployed.  Servlet container must be stopped before deploying the new `war` file.
+* This upgrade will remove FastQC resuts from the database and move them to the file system.  It is **strongly** recommended to make a backup of your database before this upgrade.
```

Comment:
So, I was thinking about this and I'm wondering if we should add a description of exactly what this update is doing to our documentation/install guide? That is, we should have a section (maybe in the FAQ, maybe in some other section) describing how this update is moving the fastqc images from the database into the filesystem and storing them in the `output-files/` directory.

The reason I think we should do this is that it's a fairly large refactor of our database, and with many groups getting into using IRIDA I feel we have an obligation to be more open about what these larger updates are doing. Especially if people, like us, are wanting to use IRIDA as there main storage system for genomics data (we don't want to have people accidentally breaking something).

In particular, for this section in our documentation, I believe we should both describe what this update is doing, as well as methods to recover from a bad update. For methods to recover, these could be divided up into 3 parts:

1. Before running FastQC migration code: restore from a back of the database.
2. In middle of writing files to temporary folder: restore database and delete temp folder.
3. After starting to move files into `output-files/` directory: restore database and delete the appropriate analysis files from the `output-files/` directory.

Once this document is written, you can link to it in the `UPGRADING.md` file here.

What are your thoughts on this? Does it make sense?

Reply:
Sure.  I think that's a good document to get started anyway in case we do more major upgrades.  Done in 7cbe0b73f5

Reply:
Thanks @tom114. This looks amazing and is exactly what I wanted.